src/http/conn.rs

```diff
@@ -490,7 +497,7 @@ enum Reading {
 
 enum Writing<B> {
     Init,
-    Body(Encoder, Option<Cursor<B>>),
+    Body(Encoder, Option<Cursor<B>>, BytesMut),
```

Comment:
Hey yeah, the only reason I wasn't using `io.write_buf` is because I'm not sure when the `io.flush` method gets called.

If you're cool with `.encode` extending `io.write_buf` and then `io.flush` inside conn, that I can do.

I'm happy with it any way, just trying to remove the extra allocations inside `.encode` if possible.